Source/Core/InputCommon/ControllerInterface/DualShockUDPClient/DualShockUDPClient.cpp

```diff
@@ -242,32 +230,48 @@ static void HotplugThreadFunc()
       }
     }
 
+    sf::SocketSelector selector;
+    for (auto& server : s_servers)
+    {
+      selector.add(server.m_socket);
+    }
+
+    using namespace std::chrono;
+    using namespace std::chrono_literals;
+    const auto timeout = s_next_listports - SteadyClock::now();
+    const auto timeout_ms = std::max(duration_cast<milliseconds>(timeout), 1ms);
```

Comment:
Keep that old "// ReceiveWithTimeout treats a timeout of zero as infinite timeout, which we don't want" comment here